packages/sdk-ui-ts/src/token/TokenService.ts

```diff
@@ -299,7 +301,21 @@ export class TokenService {
       .replaceAll('/', '-')
       .replaceAll(' ', '-')
       .toLowerCase()
+
     const [baseTokenSymbol] = market.ticker.replace(quoteToken.symbol, '')
+
+    let tokenType
```

Comment:
Extract this to a function, `getTokenTypeFromSymbol` or something like that. I also don't like that we have to classify the `baseToken` for derivative markets; they don't have such a concept since there is no `baseDenom` for derivative markets; we've added this so that it's easier for us to render some stuff on the UI level. 

Please rethink the approach here

Reply:
@bangjelkoski , yeah, I thought about how there is not a real `baseToken` in a derivatives market as well. and you're right that we're just doing it to support the concept of a `baseAmount` in a perp market and that we can have a cleaner implementation.

for this function `getBinaryOptionsMarketWithToken`, it is not currently used in the dex, since we're not currently supporting any binaryOptions markets there.  is there a UI I can use to test a new implementation of the method? perhaps make a feature branch off of Front Runner's repo and hook up to one of their markets. don't see a link to their github repo on their site though so perhaps they're closed source atm